manual/manual_assertion.rst

```diff
@@ -0,0 +1,170 @@
+
+In quantum computing, an assertion is a predefined predicate using which can let us test whether an experimentally prepared quantum state is in a specified subspace of the state space.
+
+In addition to detecting defects, the assertion schemes in ``pytket`` automatically correct the state if there is no assertion error.
+This property can be potentially exploited to help error mitigation or nondeterministically preparing a quantum state.
+
+``pytket`` provides two ways to construct an assertion, by a projector matrix or by a set of Pauli stabilisers.
+The former can be used to assert any arbitrary subspaces but notice that we currently only support 2x2, 4x4, and 8x8 matrices.
+The latter is useful for asserting whether the prepared state lies in a subspace spanned by some stabiliser states.
+
+When applied to a circuit, the assertion is inserted as a :py:class:`CircuitBox`, and then synthesized into a set of gates and measurements by the :py:class:`DecomposeBoxes` pass. Be aware that an ancilla qubit might be required for the assertion.
+The results of these measurements will be used later on to determine the outcome of the assertion.
+
+To test the circuit, compile and process the circuit using a backend in ``pytket-extensions`` that supports mid-circuit measurement and reset (e.g. :py:class:`AerBackend` from ``pytket-qiskit``).
+Once a :py:class:`BackendResult` object is retrieved, the outcome of the assertion can be checked with the :py:meth:`get_debug_info` method.
+
+
+Projector-based
+---------------
+
+Projector-based assertion utilises the simple fact that the outcome of a projective measurement can be used to determine if a quantum state is in a specified subspace of the state space.
+The method implemented in pytket transforms any arbitrary projective measurement into measurements on the computational basis [Gushu2020]_. 
+However, this approach is not without limitations. Projectors in general require :math:`n \times n` matrices to represent; hence it becomes impractical when the size of the asserted subspace is large.
+Moreover, the transformation technique we have adapted requires synthesis for arbitrary unitary matrices. Since ``pytket`` currently only supports synthesis for 1, 2, and 3 qubit unitaries, the projectors are limited to 2x2, 4x4, and 8x8 matrices.
+
+To start asserting with a projector, one should first compute the projector matrix for the target subspace. If the rank of the projector is larger than :math:`2^{n-1}` (:math:`n` is the number of qubits), an ancilla qubit should be provided to the :py:meth:`add_assert()` method.
+A special unsupported case arises when asserting a 3-qubits subspace whose projector has a rank larger than :math:`2^{3-1}`.
+
+In the following example, we try to prepare a Bell state along with a state obtained by applying a :math:`Rx(0.3)` rotation to :math:`|0\rangle`; we then use projectors to assert whether the circuit construction is correct or not.
+
+.. jupyter-input::
```

Comment:
I think we can use `jupyter-execute` for this (always preferable, since it catches code bugs and guarantees the manual is correct going forward).